src/librustdoc/clean/types.rs

```diff
@@ -1652,7 +1649,7 @@ impl PrimitiveType {
             sym::pointer => Some(PrimitiveType::RawPointer),
             sym::reference => Some(PrimitiveType::Reference),
             kw::Fn => Some(PrimitiveType::Fn),
-            sym::never => Some(PrimitiveType::Never),
+            sym::never | sym::bang => Some(PrimitiveType::Never),
```

Comment:
Unfortunately, I'm pretty sure this is still incorrect. `sym::bang == "bang"`; notice that it's in the `Symbols {` section of `rustc_span::symbol`.

Reply:
I'll just remove this change as it's unneeded in this PR.